File: VkFragment/app/src/main/java/com/example/vkontakte/ListAdapter.java
package com.example.vkontakte;

import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.media.Image;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import com.squareup.picasso.Picasso;


import java.util.List;


public class ListAdapter extends RecyclerView.Adapter<ListAdapter.ViewHolder> {

    public List<ListItem> listItems;
    public Context context;

    public ListAdapter(List<ListItem> listItems, Context context) {
        this.listItems = listItems;
        this.context = context;
    }
    @NonNull
    @Override
    public ViewHolder onCreateViewHolder( ViewGroup parent, int viewType){
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.news_items, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, int position){
        final ListItem listItem = listItems.get(position);
        holder.tvGroup.setText(listItem.getGroupName());
        holder.tvHeading.setText(listItem.getHeading());
        holder.tvLikes.setText(listItem.getLikes());
        holder.tvComments.setText(listItem.getComments());
        holder.tvViews.setText(listItem.getViews());
        holder.tvTime.setText(listItem.getPublishDate());
        holder.tvShares.setText(listItem.getShares());

        Picasso.with(context)
        .load(listItem.getImgURL())
        .into(holder.ivPost);

        Picasso.with(context)
                .load(listItem.getImgURL())
                .into(holder.ivGroupPhoto);


        holder.ibLikes.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view){
                Toast.makeText(context, "You liked this post ", Toast.LENGTH_LONG).show();
                holder.tvLikes.setText(Integer.toString(Integer.parseInt(listItem.getLikes()) + 1));
            }
        });

        holder.setItemClickListener(new ItemClickListener() {
            @Override
            public void onItemClickListener(View v, int position) {
                String tvGroup = listItems.get(position).getGroupName();
                String tvHeading = listItems.get(position).getHeading();
                String tvLikes = listItems.get(position).getLikes();
                String tvComments = listItems.get(position).getComments();
                String tvShares = listItems.get(position).getShares();
                String tvTime = listItems.get(position).getPublishDate();
                String tvViews = listItems.get(position).getViews();
                String imgURL = listItems.get(position).getImgURL();

                Picasso.with(context)
                        .load(imgURL)
                        .into(holder.ivPost);

                Picasso.with(context)
                        .load(imgURL)
                        .into(holder.ivGroupPhoto);


                Intent intent = new Intent(context, ClickActivity.class);
                intent.putExtra("tvGroup", tvGroup);
                intent.putExtra("tvHeading", tvHeading);
                intent.putExtra("tvLikes", tvLikes);
                intent.putExtra("tvComments", tvComments);
                intent.putExtra("tvShares", tvShares);
                intent.putExtra("tvTime", tvTime);
                intent.putExtra("tvViews", tvViews);
                intent.putExtra("imgURL", imgURL);

                context.startActivity(intent);
            }

        });


    }

    @Override
    public int getItemCount(){
        return listItems.size();
    }



    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{

        TextView tvGroup, tvTime, tvHeading, tvLikes, tvComments, tvShares, tvViews;
        ImageView ivPost, ivGroupPhoto;
        CardView cardView;
        ImageButton ibLikes;
        ItemClickListener itemClickListener;


        ViewHolder(@NonNull View itemView) {
            super(itemView);
            tvGroup = itemView.findViewById(R.id.tvGroup);
            tvTime =  itemView.findViewById(R.id.tvTime);
            tvHeading =  itemView.findViewById(R.id.tvHeading);
            tvLikes = itemView.findViewById(R.id.tvLikes);
            tvComments =itemView.findViewById(R.id.tvComment);
            tvViews = itemView.findViewById(R.id.tvViews);
            ivPost = itemView.findViewById(R.id.ivPost);
            ivGroupPhoto = itemView.findViewById(R.id.ivGroupPhoto);
            cardView = itemView.findViewById(R.id.cardView);
            ibLikes = itemView.findViewById(R.id.iblike);
            tvShares = itemView.findViewById(R.id.tvShare);

            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            this.itemClickListener.onItemClickListener(v, getLayoutPosition());
        }

        public void setItemClickListener(ItemClickListener ic){
            this.itemClickListener = ic;
        }
    }
}

File: ktVkontakte/app/src/main/java/com/example/ktvkontakte/ListAdapter.kt
package com.example.ktvkontakte

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.squareup.picasso.Picasso

//class ListAdapter(var list: List<ListItem>,
//val itemClickListener: RecyclerViewItemClick? = null) :
//    RecyclerView.Adapter<ListAdapter.ViewHolder>() {
//
//
//    inner class ViewHolder(val view: View) : RecyclerView.ViewHolder(view) {
//        fun bind(post: ListItem) {
//
//            val tvGroup = itemView.findViewById<TextView>(R.id.tvGroup)
//            val tvTime = itemView.findViewById<TextView>(R.id.tvTime)
//            val tvHeading = itemView.findViewById<TextView>(R.id.tvHeading)
//            val tvLikes = itemView.findViewById<TextView>(R.id.tvLikes)
//            val tvComments = itemView.findViewById<TextView>(R.id.tvComment)
//            val tvViews = itemView.findViewById<TextView>(R.id.tvViews)
//            val ivPost = itemView.findViewById<ImageView>(R.id.ivPost)
//            val ivGroupPhoto = itemView.findViewById<ImageView>(R.id.ivGroupPhoto)
//            val cardView = itemView.findViewById<View>(R.id.cardView)
//            val ibLikes = itemView.findViewById<ImageButton>(R.id.iblike)
//            val tvShares = itemView.findViewById<TextView>(R.id.tvShare)
//
//            tvGroup.text = post.groupName;
//            tvHeading.text = post.heading;
//            tvLikes.text = post.likes;
//            tvComments.text = post.comments;
//            tvViews.text = post.views;
//            tvTime.text = post.publishDate;
//            tvShares.text = post.shares;
//
//            Picasso.with(view.context)
//                .load(post.imgURL)
//                .into(ivPost)
//
//            Picasso.with(view.context)
//                .load(post.imgURL)
//                .into(ivGroupPhoto)
//
//            view.setOnClickListener {
//                itemClickListener?.itemClick(adapterPosition, post)
//            }
//
//        }
//    }
//
//
//
//    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
//        val view = LayoutInflater.from(parent.context).inflate(R.layout.news_items, parent, false)
//        return ViewHolder(view)
//    }
//
//    override fun getItemCount(): Int = list.size ?: 0
//
//
//    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
//        holder.bind(list[position])
//    }
//
//    interface RecyclerViewItemClick {
//        fun itemClick(position: Int, item: ListItem)
//    }
//}









































class ListAdapter(var listItems: List<ListItem>,
                  val itemClickListener: RecyclerViewItemClick? = null) :
    RecyclerView.Adapter<ListAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListAdapter.ViewHolder {
        val v: View = LayoutInflater.from(parent.context).inflate(R.layout.news_items, parent, false)
        return ViewHolder(v)
    }

    override fun getItemCount(): Int = listItems.size ?: 0

    override fun onBindViewHolder(holder: ListAdapter.ViewHolder, position: Int) {
        holder.bind(listItems[position])
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bind(post: ListItem) {
            val tvGroup = itemView.findViewById<TextView>(R.id.tvGroup)
            val tvTime = itemView.findViewById<TextView>(R.id.tvTime)
            val tvHeading = itemView.findViewById<TextView>(R.id.tvHeading)
            val tvLikes = itemView.findViewById<TextView>(R.id.tvLikes)
            val tvComments = itemView.findViewById<TextView>(R.id.tvComment)
            val tvViews = itemView.findViewById<TextView>(R.id.tvViews)
            val ivPost = itemView.findViewById<ImageView>(R.id.ivPost)
            val ivGroupPhoto = itemView.findViewById<ImageView>(R.id.ivGroupPhoto)
            val ibLikes = itemView.findViewById<ImageButton>(R.id.iblike)
            val tvShares = itemView.findViewById<TextView>(R.id.tvShare)

            tvGroup.text = post.groupName
            tvHeading.text = post.heading
            tvLikes.text = post.likes
            tvComments.text = post.comments
            tvViews.text = post.views
            tvTime.text = post.publishDate
            tvShares.text = post.shares

            Picasso.with(itemView.context)
                .load(post.imgURL)
                .into(ivPost)

            Picasso.with(itemView.context)
                .load(post.imgURL)
                .into(ivGroupPhoto)


            ibLikes.setOnClickListener {
                Toast.makeText(itemView.context, "You liked this post", Toast.LENGTH_SHORT).show()
                tvLikes.text = (tvLikes.text).toString().toInt().plus(1).toString()
            }

            itemView.setOnClickListener {
                itemClickListener?.itemClick(adapterPosition, post)
            }
        }
    }

    interface RecyclerViewItemClick {
        fun itemClick(position: Int, item: ListItem)
    }
}